src/server/transaction/cmds/cmds.go

```diff
@@ -0,0 +1,272 @@
+package cmds
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/pkg/grpcutil"
+	"github.com/pachyderm/pachyderm/src/client/transaction"
+	"github.com/pachyderm/pachyderm/src/server/pkg/cmdutil"
+	"github.com/pachyderm/pachyderm/src/server/pkg/tabwriter"
+	"github.com/pachyderm/pachyderm/src/server/transaction/pretty"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// Cmds returns the set of commands used for managing transactions with the
+// Pachyderm CLI tool pachctl.
+func Cmds(noMetrics *bool, noPortForwarding *bool) []*cobra.Command {
+	var commands []*cobra.Command
+
+	marshaller := &jsonpb.Marshaler{Indent: "  "}
+
+	raw := false
+	rawFlags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	rawFlags.BoolVar(&raw, "raw", false, "disable pretty printing, print raw json")
+
+	fullTimestamps := false
+	fullTimestampsFlags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	fullTimestampsFlags.BoolVar(&fullTimestamps, "full-timestamps", false, "Return absolute timestamps (as opposed to the default, relative timestamps).")
+
+	transactionDocs := &cobra.Command{
+		Short: "Docs for transactions.",
+		Long:  "Do a few thing.",
+	}
+	cmdutil.SetDocsUsage(transactionDocs)
+	commands = append(commands, cmdutil.CreateAlias(transactionDocs, "transaction"))
+
+	listTransaction := &cobra.Command{
+		Short: "List transactions.",
+		Long:  "List transactions.",
+		Run: cmdutil.RunFixedArgs(0, func([]string) error {
+			c, err := client.NewOnUserMachine(!*noMetrics, !*noPortForwarding, "user")
+			if err != nil {
+				return err
+			}
+			defer c.Close()
+			transactions, err := c.ListTransaction()
+			if err != nil {
+				return err
+			}
+			if raw {
+				for _, transaction := range transactions {
+					if err := marshaller.Marshal(os.Stdout, transaction); err != nil {
+						return err
+					}
+				}
+				return nil
+			}
+			writer := tabwriter.NewWriter(os.Stdout, pretty.TransactionHeader)
+			for _, transaction := range transactions {
+				pretty.PrintTransactionInfo(writer, transaction, fullTimestamps)
+			}
+			return writer.Flush()
+		}),
+	}
+	listTransaction.Flags().AddFlagSet(fullTimestampsFlags)
+	commands = append(commands, cmdutil.CreateAlias(listTransaction, "list transaction"))
+
+	startTransaction := &cobra.Command{
+		Short: "Start a new transaction.",
+		Long:  "Start a new transaction.",
+		Run: cmdutil.RunFixedArgs(0, func([]string) error {
+			c, err := client.NewOnUserMachine(!*noMetrics, !*noPortForwarding, "user")
+			if err != nil {
+				return err
+			}
+			defer c.Close()
+			transaction, err := c.StartTransaction()
+			if err != nil {
+				return grpcutil.ScrubGRPC(err)
+			}
+			// TODO: use advisory locks on config so we don't have a race condition if
+			// two commands are run simultaneously
```

Comment:
This will be done in a future PR - the worst that can happen from this is users mix up two transactions.